Explain how to setup a development environment (#597)
grammyjs/website: site/docs/guide/introduction.md

### Prerequisites for Deno

Before you can start creating your bot, let's first spend a few minutes on a proper setup for software development.
This means installing a few tools.

#### Preparing Your Machine for Development

[Install Deno](https://deno.land/manual/getting_started/installation#download-and-install) if you have not done that already.

> You also need a text editor that works well for coding.

Reviewer: ```suggestion
You also need a text editor that fits well with coding.
```

The best one for Deno is Visual Studio Code.
[Install it.](https://code.visualstudio.com/)

> Next, you need to connect VSCode and Deno.

Reviewer: ```suggestion
Next, you need to connect Visual Studio Code and Deno.
```

Author: We are already using VSCode in more places in the docs and I don't mind. It's the more common term and it's more readable, too. I find this a good place to implicitly introduce this.

Do you maybe want to introduce the term explicitly instead, assuming that people don't get what we're talking about?

Reviewer: But is it VSCode or VS Code?

Reviewer: Notice https://github.com/microsoft/vscode#related-projects

Author: We can do VS Code, but we'll have to do it consistently everywhere. We currently use VSCode at least once.

Reviewer: OK. In another PQ.

> That's very simple: there is an extension for VSCode that does everything automatically.

Reviewer: ```suggestion
That's very simple: there is an extension for Visual Studio Code that does everything automatically.
```

You can install it [as described here](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno).

Your system is now ready for bot development! :tada:

#### Developing a Bot

Create a new directory somewhere.
It will contain your bot project.
Open this new directory in VSCode.

```sh
mkdir ./my-bot
cd ./my-bot
code .
```

> 

Reviewer: ```suggestion
> If the `code` command is not available, just open VSCode and open your directory from File > Open Folder...

```

Author: Does this ever happen? 🫠

Reviewer: Yes, it’s not there by default on macOS.

Author: wtf

In VSCode, turn this empty directory into a Deno project.

> Hit `Ctrl+Shift+P`, type "deno init" and hit Enter.

Reviewer: ```suggestion
Hit `Ctrl+Shift+P`, type "deno init", and hit Enter.
```

> The bottom right of your editor should now display the Deno version on your system.

Reviewer: ```suggestion
The bottom right of your editor should now display the version of Deno installed on your system.
```

> Your Deno project is now ready so you can create your bot.

Reviewer: ```suggestion
Your Deno development environment should now be ready. So, you can now start writing your bot.
```

Author: ```suggestion
Your Deno development environment is ready.
You can now create your bot.
```
?

Reviewer: Better. But I still prefer write instead of create.

Author: I missed that part

Author: ```suggestion
Your Deno development environment is ready.
You can now start writing your bot.
```
?

### Prerequisites for Deno

Before you can start creating your bot, let's first spend a few minutes on a proper setup for software development.
This means installing a few tools.

#### Preparing Your Machine for Development

[Install Deno](https://deno.land/manual/getting_started/installation#download-and-install) if you have not done it already.

You also need a text editor that fits well with coding.
The best one for Deno is Visual Studio Code.
[Install it.](https://code.visualstudio.com/)

Next, you need to connect VSCode and Deno.
That's very simple: There is an extension for VSCode that does everything automatically.
You can install it [as described here](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno).

Your system is now ready for bot development! :tada:

#### Developing a Bot

Create a new directory somewhere.
It will contain your bot project.
Open this new directory in VSCode.

```sh
mkdir ./my-bot
cd ./my-bot
code .
```

> > If the `code` command is not available, just open VSCode and open your directory from File > Open Folder...

Author: Perhaps we should rather explain how to fix the setup on Apple devices? I don't want to bother explaining Windows, but for Mac we could at least try

Reviewer: Done.

> > If the `code` command is not available, just open VSCode and open your directory from File > Open Folder...

Reviewer: ```suggestion

> If you’re on macOS, and the `code` command is not available, just open VSCode, hit `Ctrl+Shift+P`, type "shell command", and hit enter.
```

Author: Love it.

### Prerequisites for Deno

Before you can start creating your bot, let's first spend a few minutes on a proper setup for software development.
This means installing a few tools.

#### Preparing Your Machine for Development

[Install Deno](https://deno.land/manual/getting_started/installation#download-and-install) if you have not done it already.

You also need a text editor that fits well with coding.
The best one for Deno is Visual Studio Code.
[Install it.](https://code.visualstudio.com/)

Next, you need to connect VSCode and Deno.
That's very simple: There is an extension for VSCode that does everything automatically.
You can install it [as described here](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno).

Your system is now ready for bot development! :tada:

#### Developing a Bot

Create a new directory somewhere.
It will contain your bot project.
Open this new directory in VSCode.

```sh
mkdir ./my-bot
cd ./my-bot
code .
```

> > If you’re on macOS and the `code` command is not available, just open VSCode, hit `Ctrl+Shift+P`, type "shell command", and hit enter.

Author: Do you have any idea how this works on Windows?

Reviewer: No. I wouldn’t bother tho. There’s something called PoawaerShall if I’m not mistaken. It should probably be there. Otherwise let them use WSL, or grab a VMWare torrent if they’re on Windows 7 or older.

Author: Alright, then let's not think about it

Reviewer: Happy :)

### Prerequisites for Deno

Before you can start creating your bot, let's first spend a few minutes on a proper setup for software development.
This means installing a few tools.

#### Preparing Your Machine for Development

[Install Deno](https://deno.land/manual/getting_started/installation#download-and-install) if you have not done it already.

You also need a text editor that fits well with coding.

> The best one for Deno is Visual Studio Code.
> [Install it.](https://code.visualstudio.com/)
> 
> Next, you need to connect VS Code and Deno.

Reviewer 2: ```suggestion
The best one for Deno is Visual Studio Code (VS Code).
[Install it.](https://code.visualstudio.com/)

Next, you need to connect VS Code and Deno.
```

Author: ```suggestion
The best one for Deno is Visual Studio Code, often just called VS Code.
[Install it.](https://code.visualstudio.com/)

Next, you need to connect VS Code and Deno.
```

That's very simple: There is an extension for VSCode that does everything automatically.
You can install it [as described here](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno).

Your system is now ready for bot development! :tada:

#### Developing a Bot

Create a new directory somewhere.
It will contain your bot project.
Open this new directory in VSCode.

```sh
mkdir ./my-bot
cd ./my-bot
code .
```

> > If you’re on macOS and the `code` command is not available, just open VSCode, hit `Ctrl+Shift+P`, type "shell command", and hit enter.

Reviewer: ```suggestion
> If you’re on macOS and the `code` command is not available, just open VSCode, hit `Ctrl+Shift+P`, type "shell command", and hit Enter.
```

> > If you’re on macOS and the `code` command is not available, just open VS Code, hit `Ctrl+Shift+P`, type "shell command", and hit Enter.

Reviewer 2: I'm not so familiar with macOS. I believe the CTRL key in windows is CMD in macOS?
```suggestion
> If you’re on macOS and the `code` command is not available, just open VS Code, hit `Cmd+Shift+P`, type "shell command", and hit Enter.
```

Author: This one is for @roj1512 to decide

Reviewer: Thanks for correcting me, @CikiMomogi. I’d even put the command icon.

Reviewer 2: Btw, we can use the kbd tag to make them look more stylish.

![image](https://user-images.githubusercontent.com/74030149/218291293-aa5aeb82-36f2-49dc-a2c8-7d16f33425b6.png)
